src/unix/glegl.cpp

```diff
@@ -364,9 +364,51 @@ bool wxGLCanvasEGL::InitVisual(const wxGLAttributes& dispAttrs)
     return m_config != nullptr;
 }
 
+
+extern "C"
+{
+
+typedef EGLDisplay (*GetPlatformDisplayFunc)(EGLenum platform,
+                                             void* native_display,
+                                             const EGLAttrib* attrib_list);
+
+// Returns the appropriate function pointer for "eglGetPlatformDisplay()" or
+// "eglGetPlatformDisplayEXT()" or nullptr if neither is available at runtime.
+static GetPlatformDisplayFunc get_edl_platform_display_func()
+{
+    // Static initializer ensures this is only computed once.
+    static const GetPlatformDisplayFunc s_getPlatformDisplayFunc([] ()
+    {
+        const GetPlatformDisplayFunc getPlatformDisplayFunc =
+            reinterpret_cast<GetPlatformDisplayFunc>(
+                eglGetProcAddress("eglGetPlatformDisplay"));
```

Comment:
It looks like just calling `eglGetProcAddress()` might not be enough.  :-(  From the [documentation](https://registry.khronos.org/EGL/sdk/docs/man/html/eglGetProcAddress.xhtml):

> A non-NULL return value does not guarantee that an extension function is actually supported at runtime. The client must also make a corresponding query, such as glGetString(GL_EXTENSIONS) for OpenGL and OpenGL ES extensions; vgGetString(VG_EXTENSIONS) for OpenVG extensions; eglQueryString(display, EGL_EXTENSIONS); or query the EGL or client API version for non-extension functions, to determine if a function is supported by EGL or a specific client API context.

So it seems it might be necessary to call `eglQueryString` also.

Reply:
Here is how GTK seems to do it, for reference:
https://gitlab.gnome.org/GNOME/gtk/-/blob/main/gdk/gdkdisplay.c#L1488

Reply:
I really think the `EGL_KHR_platform_base` part is incorrect. I can at least attest of one 1.5 EGL implementation that doesn't use it. It is also nowhere to be found in the EGL documentation.

Reply:
> I really think the `EGL_KHR_platform_base` part is incorrect. I can at least attest of one 1.5 EGL implementation that doesn't use it. It is also nowhere to be found in the EGL documentation.

You might be right about `EGL_KHR_platform_base`.  [This](https://registry.khronos.org/EGL/api/egl.xml) seems to suggest that `EGL_EXT_platform_base` should be used instead.

Reply:
So is the current version ok to merge, @swt2c?

Reply:
I still think that checks for `EGL_EXT_platform_base` in the query string should be added too.

Also fix the presumed typo I pointed out.

Reply:
I am now querying for `EGL_EXT_platform_base` before trying `eglGetProcAddress()`. Please, let me know what you think.

Reply:
This looks OK to me now.

My only nitpick would be the lambda function as this can't be backported as-is, but it's not a dealbreaker.  :-)